fe/fe-core/src/main/java/com/starrocks/sql/ast/StructFieldDesc.java

```diff
@@ -0,0 +1,154 @@
+    }
+
+    @Override
+    public NodePosition getPos() {
+        return null;
+    }
+
+    private boolean checkType(Type type) {
+        if (type.isStructType() || type.isArrayType()) {
+            return true;
+        }
+        return false;
+    }
+
+    private Type getFieldType(Type type, String fieldName) {
+        if (type.isStructType()) {
+            StructField field = ((StructType) type).getField(fieldName);
+            if (field == null) {
+                return null;
+            }
+            return field.getType();
+        }
+
+        if (type.isArrayType()) {
+            return ((ArrayType) type).getItemType();
+        }
+
+        return null;
+    }
+
+    public void analyze(Column baseCol, boolean dropField) throws AnalysisException {
+        if (baseCol == null) {
+            throw new AnalysisException(String.format("Analyze add/drop field failed, modifyc column is not exist"));
+        }
+
+        Type targetFieldType = baseCol.getType();
+        if (!checkType(targetFieldType)) {
+            throw new AnalysisException(
+                    String.format("column %s type %s is not Struct", baseCol.getName(), targetFieldType.toString()));
+        }
+
+        if (nestedParentFieldNames != null && !nestedParentFieldNames.isEmpty()) {
+            for (String name : nestedParentFieldNames) {
+                targetFieldType = getFieldType(targetFieldType, name);
+                if (targetFieldType == null) {
+                    throw new AnalysisException(
+                        String.format("No field %s exist in column %s", name, baseCol.getName()));
+                }
+                if (!checkType(targetFieldType)) {
+                    throw new AnalysisException(
+                        String.format("Field %s type %s is not valid", name, targetFieldType.toString()));
+                }
+            }
+        }
+
+        if (!targetFieldType.isStructType()) {
+            throw new AnalysisException("Target Field is not struct");
+        }
+
+        StructField childField = ((StructType) targetFieldType).getField(fieldName);
+        if (dropField) {
+            if (childField == null) {
+                throw new AnalysisException(String.format("Drop field %s is not found", fieldName));
+            }
+        } else {
+            if (childField != null) {
+                throw new AnalysisException(String.format("Field %s is already exist", fieldName));
+            }
+
+            if (typeDef == null) {
+                throw new AnalysisException("No filed type in field definition");
+            }
+            if (typeDef.getType().isScalarType()) {
+                final ScalarType targetType = (ScalarType) typeDef.getType();
+                if (targetType.getPrimitiveType().isStringType()) {
+                    if (targetType.getLength() <= 0) {
+                        targetType.setLength(1);
+                    }
+                }
+            }
```

Comment:
child field may be struct type？

Reply:
> child field may be struct type？

Yes, child filed may be struct or array type.